### PR TITLE
PE-1919: Refactor custom Arweave Gateway URL

### PR DESCRIPTION
**Pull request by matibat**
Changes

- Uses `LocalKeyValueStore` instead of a cookie to hold the custom URL
- Removes the `cooky` dep
  - ~FIXME: Some versions were unintentionally updated~
- Adds an entry to the README.md explaining the new behavior

### CODE REVIEW

File: lib/utils/local_key_value_store.dart [hunk 1]
```diff
@@ -31,8 +31,23 @@ class LocalKeyValueStore implements KeyValueStore {
...
+    return _preferences.getString(key);
+  }
+
   @override
   Future<bool> remove(String key) {
     return _preferences.remove(key);
   }
+
+  @override
+  Set<String> getKeys() {
```
Comment (matibat) [inline lib/utils/local_key_value_store.dart:50]: TODO: add unit tests for the newly implemented methods

Comment (matibat) [reply lib/utils/local_key_value_store.dart:50]: ea7420e97fe76f5e31c1152c455d7662188bb616

File: README.md [hunk 2]
```diff
@@ -56,3 +56,19 @@ All changes made to `dev` will be continuously deployed to [staging.ardrive.io](
 To create a release to [app.ardrive.io](https://app.ardrive.io), first merge any changes from `dev` into `master` that are required, and publish a new release through the GitHub UI with the tag name matching the pattern `v*` eg. `v1.0.1`.
 
 This will trigger a GitHub Action that will deploy `master` to production.
+
+## Using a custom Arweave Gateway URL
+
+You can use a custom gateway URL by setting the `flutter.arweaveGatewayUrl` item in your local storage.
```
Comment (karlprieb) [inline README.md:62]:
```suggestion
You can use a custom gateway URL by setting the `flutter.arweaveGatewayUrl` key in browser's Local Storage.
```

File: README.md [hunk 3]
```diff
@@ -56,3 +56,19 @@ All changes made to `dev` will be continuously deployed to [staging.ardrive.io](
...
+
+In web you can use this example in the JavaScript console of your browser:
```
Comment (karlprieb) [inline README.md:64]:
```suggestion
With the web app opened you can access the browser JavaScript console to run the follow examples:
```

File: lib/services/config/config_service.dart [hunk 4]
```diff
@@ -11,14 +11,15 @@ class ConfigService {
 
   Future<AppConfig> getConfig() async {
     if (_config == null) {
-      final environment = kReleaseMode ? 'prod' : 'dev';
+      const environment = kReleaseMode ? 'prod' : 'dev';
       final configContent =
           await rootBundle.loadString('assets/config/$environment.json');
 
-      final gatewayCookie = cookie.get('arweaveGatewayUrl');
+      final localStorage = await LocalKeyValueStore.getInstance();
```
Comment (thiagocarvalhodev) [inline lib/services/config/config_service.dart:18]: Can we pass this instance by dependency injection? So if we want to change the `KeyValueStore` we can do without changing its class.

**Review by thiagocarvalhodev — APPROVED** [review]
Looking good! Just left a single comment.

File: lib/services/config/config_service.dart (hunk 4 above)
Comment (matibat) [reply lib/services/config/config_service.dart:18]: The instance is currently a singleton. We use dependency injection, but for tests only

Comment (thiagocarvalhodev) [reply lib/services/config/config_service.dart:18]: I mean, pass it for dependency injection to `ConfigService`.

Comment (matibat) [reply lib/services/config/config_service.dart]: Ohh I misunderstood. So yea, what you said makes total sense: 2ec310c8f1449b17f6e9a44bd2e3dac8d978f455

Comment (matibat) [reply lib/services/config/config_service.dart]: That changed the tests as well: f07770a753784c8a87a73b63dd5657dcabc2dd97

**Comment by thiagocarvalhodev** [comment]
LGTM!

**Review by thiagocarvalhodev — APPROVED** [review]

**Review by fedellen — APPROVED** [review]

**Review by karlprieb — APPROVED** [review]

**Pull request merged**